File: Python/programmers/etc/level2/하노이의_탑.py
def solution(n):
    answer = []
    # 시작, 보조, 끝
    def dfs(n, fr, aux, to):
        if n == 1:
            answer.append([fr, to])
            return
        # (n-1)개 보조 기둥으로 이동
        dfs(n-1, fr, to, aux)
        # 가장 큰 거 끝으로 옮기고
        answer.append([fr, to])
        # (n-1)개 끝으로 이동하면 끝
        dfs(n-1, aux, fr, to)
    
    dfs(n, 1, 2, 3)
    
    return answer


n = 2
print(solution(n)) # [[1,2], [1,3], [2,3]]

File: Java/programmers/etc/level2/하노이의탑.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 하노이의탑 {
    static List<int[]> answer = new ArrayList<>();

    public List<int[]> solution(int n) {
        dfs(n, 1, 2, 3);
        return answer;
    }

    private static void dfs(int n, int fr, int aux, int to) {
        if (n == 1) {
            answer.add(new int[]{fr, to});
            return;
        }
        dfs(n-1, fr, to, aux);
        answer.add(new int[]{fr, to});
        dfs(n-1, aux, fr, to);
    }
}
